src/pretix/presale/views/order.py

```diff
@@ -602,13 +602,18 @@ def get_confirm_url(self, payment):
 
 
 def can_generate_invoice(event, order, ignore_payments=False):
+    open_payment = order.payments.last()
+    if open_payment and open_payment.state in (OrderPayment.PAYMENT_STATE_CONFIRMED, OrderPayment.PAYMENT_STATE_REFUNDED):
+        open_payment = None
```

Comment:
I'm unsure about the semantics here. This is taken from the `OrderChangeManager`.
I thought Payments in state `created` and `pending` would be "open". Either way, we need to decide in which payment states invoices should be able to be created for payment providers overwriting this option.

Reply:
Hmm, you're right, this logic kinda handles failed/canceled payments wrong. However, for the scope of this PR, I think we can drop this again and automatically create the invoice when the payment method has changed, so we don't need to touch this method at all.